WalletWasabi/Blockchain/Keys/HdPubKey.cs

```diff
@@ -97,10 +97,14 @@ public int AnonymitySet
 
 		private int HashCode { get; }
 
-		public void SetAnonymitySet(int anonset, uint256 reason)
+		public void SetAnonymitySet(int anonset, uint256 reason, bool isOutput)
 		{
+			if (isOutput)
+			{
+				OutputAnonSetReasons.Add(reason);
+			}
+
```

Comment:
Try to avoid boolean parameters. When you see a method like `DoSomething(true)` you know that there is an `if` statement to do one thing when `true` and something else when `false`. In these cases is use to be better to have two different methods `DoSomething()` and `DoSomethingElse()` because it is immediately clear what they do.

In this case I would suggest to remove boolean as follow: 

```suggestion
		public void SetAnonymitySet(int anonset, uint256 reason = null)
		{
			if (reason is {})
			{
				OutputAnonSetReasons.Add(reason);
			}

```

Reply:
So if I understand it correctly, we should remove the boolean, and in that case if we dont want to add the reason, we should call it like ```SetAnonimitySet(anonset)``` ?
Wouldn't it cause null reference errors?

Reply:
No. That's why there is a null check `if (reason is { })`